src/process.rs

```diff
@@ -278,6 +290,43 @@ fn read_process_availabilities(
         .with_context(|| input_err_msg(&file_path))
 }
 
+fn read_process_flows_from_iter<I>(
+    iter: I,
+    process_ids: &HashSet<Rc<str>>,
+    commodities: &HashMap<Rc<str>, Rc<Commodity>>,
+) -> Result<HashMap<Rc<str>, Vec<ProcessFlow>>>
+where
+    I: Iterator<Item = ProcessFlowRaw>,
+{
+    iter.map(|flow_raw| -> Result<ProcessFlow> {
+        let commodity = commodities
+            .get(flow_raw.commodity_id.as_str())
+            .with_context(|| format!("{} is not a valid commodity ID", &flow_raw.commodity_id))?;
+
+        Ok(ProcessFlow {
+            process_id: flow_raw.process_id,
+            commodity: Rc::clone(commodity),
+            flow: flow_raw.flow,
+            flow_type: flow_raw.flow_type,
+            flow_cost: flow_raw.flow_cost,
+        })
+    })
+    .collect::<Result<Vec<_>>>()?
+    .into_iter()
+    .into_id_map(process_ids)
```

Comment:
If you do it this way, you can avoid allocating a `Vec` just to throw it away again:

```suggestion
    .process_results(|iter| iter.into_id_map(process_ids))?
```